js/customize-partial-refresh-settings.js

```diff
@@ -0,0 +1,128 @@
+/* global jQuery, _customizePartialRefreshSettings, JSON, _ */
+/* exported customizePartialRefreshSettings */
+var customizePartialRefreshSettings = ( function( $ ) {
+	'use strict';
+
+	var self, api = wp.customize;
+
+	self = {
+		setting: null
+	};
+
+	$.extend( self, _customizePartialRefreshSettings );
+
+	/**
+	 * Inject the functionality.
+	 */
+	self.init = function() {
+		api.bind( 'ready', function() {
+			self.onChangeSetting();
+		} );
+
+		api.bind( 'setting-partial', function( response ) {
+			var that = $( 'iframe' ).contents(),
+				value = $.trim( response );
+
+			$( self.setting.selector, that ).html( value );
+		} );
+	};
+
+	/**
+	 * Change event for selective settings, requests a new value.
+	 */
+	self.onChangeSetting = function() {
+		$.each( self.settings, function( id, selector ) {
+
+			api( id, function( setting ) {
+
+				setting.bind( function() {
+					self.setting = {
+						id: id,
+						selector: selector
+					};
+					self.request();
+				} );
+			} );
+		} );
+	};
+
+	/**
+	 * Request a new setting value.
+	 *
+	 * @return {jQuery.Deferred}
+	 */
+	self.request = function() {
+		var spinner = $( '#customize-header-actions .spinner' ),
+			active = 'is-active',
+			deferred = $.Deferred(),
+			req = self.debounceRequest();
+
+		spinner.addClass( active );
+
+		req.done( function( response ) {
+			deferred.resolve();
+			api.trigger( 'setting-partial', response );
+			spinner.removeClass( active );
+		} );
+
+		req.fail( function() {
+			deferred.reject.apply( deferred, arguments );
+			api.previewer.refresh();
+			spinner.removeClass( active );
+		} );
+
+		return deferred;
+	};
+
+	/**
+	 * Debounce the requests, allowing setting changes made back-to-back to be sent together.
+	 *
+	 * @return {jQuery.Deferred}
+	 */
+	self.debounceRequest = ( function() {
+		var request, debouncedDeferreds = [];
+
+		request = _.debounce( function() {
+			var req, dirtyCustomized = {};
+
+			api.each( function( value, key ) {
+				if ( value._dirty ) {
+					dirtyCustomized[ key ] = value();
+				}
+			} );
+
+			req = wp.ajax.post( self.action, {
```

Comment:
> Second, even though the Ajax requests are debouncing and working as expected, if you rapid fire text changes they tend to make the preview out of sync. So the initial ajax request should have a timeout on it to slow down the responsiveness by maybe 250 milliseconds, which should allow the requests to properly debounce for things so trivial as a text chance. 

We can alleviate this by keeping track of any existing requests that are open for a given setting and doing `req.abort()` to make sure that its success callbacks never fire. This was implemented for widgets in the customizer: https://github.com/xwp/wordpress-develop/blob/baaa8946a22d97448c7e41be2f0df21858d770f6/src/wp-admin/js/customize-widgets.js#L1131-L1133